web-services/query/src/main/java/datawave/webservice/query/runner/RunningQuery.java

```diff
@@ -285,41 +292,42 @@ public ResultsPage next() throws Exception {
+                    if (o instanceof DefaultEvent) {
+                        if (((DefaultEvent) o).isIntermediateResult()) {
+                            hitIntermediateResult = true;
+                            break;
+                        }
                     }
-                }
-                
-                // regardless whether the transform iterator returned a result, it may have updated the metrics (next/seek calls etc.)
-                if (iter.getTransformer() instanceof WritesQueryMetrics) {
-                    ((WritesQueryMetrics) iter.getTransformer()).writeQueryMetrics(this.getMetric());
-                }
-                
-                // if not still waiting on a future, then process the result (or lack thereof)
-                if (future == null) {
-                    if (null == o) {
-                        log.debug("Null result encountered, no more results");
-                        this.finished = true;
-                        break;
+                    
+                    // regardless whether the transform iterator returned a result, it may have updated the metrics (next/seek calls etc.)
+                    if (iter.getTransformer() instanceof WritesQueryMetrics) {
+                        ((WritesQueryMetrics) iter.getTransformer()).writeQueryMetrics(this.getMetric());
                     }
-                    resultList.add(o);
-                    if (this.logic.getPageByteTrigger() > 0) {
-                        currentPageBytes += ObjectSizeOf.Sizer.getObjectSize(o);
+                    
+                    // if not still waiting on a future, then process the result (or lack thereof)
+                    if (future == null) {
+                        if (null == o) {
+                            log.debug("Null result encountered, no more results");
+                            this.finished = true;
+                            break;
+                        }
+                        resultList.add(o);
+                        if (this.logic.getPageByteTrigger() > 0) {
+                            currentPageBytes += ObjectSizeOf.Sizer.getObjectSize(o);
+                        }
+                        currentPageCount++;
+                        numResults++;
                     }
-                    currentPageCount++;
-                    numResults++;
+                    
+                    testForUncaughtException(resultList.size());
+                    hasNextFuture = executor.submit(() -> this.iter.hasNext());
                 }
-                
-                testForUncaughtException(resultList.size());
-                
+            } catch (TimeoutException te) {
+                // This means the iter.hasNext() call didn't return within the allotted time. Manually return an
+                // intermediate result -- don't have a count here to know how many times we've returned an intermediate result...
+                hitIntermediateResult = true;
```

Comment:
This will only get triggered by that check you were asking if we wanted hard-coded -- As I understood it, if hasNext hasn't returned in 50 minutes (regardless of whether or not this is a long running query) then we're hung.  Maybe this should just return no results instead of manually running an intermediate result since it has nothing to do with long running queries

Reply:
Done